devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -0,0 +1,550 @@
+			state "eco", label:'${name}', action:"heat", nextState:"heat", icon:"http://cdn.device-icons.smartthings.com/Outdoor/outdoor3-icn@2x.png"
+		}
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2) {
+			state "heatingSetpoint", label:'Setpoint ${currentValue}°', backgroundColors:[
+					[value: 31, color: "#153591"],
+					[value: 44, color: "#1e9cbb"],
+					[value: 59, color: "#90d2a7"],
+					[value: 74, color: "#44b621"],
+					[value: 84, color: "#f1d801"],
+					[value: 95, color: "#d04e00"],
+					[value: 96, color: "#bc2323"]
+			]
+		}
+		standardTile("refresh", "device.refresh", decoration: "flat", width: 2, height: 2) {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+
+		main ("thermostatMulti")
+		details(["thermostatMulti", "mode", "heatingSetpoint", "refresh"])
+	}
+}
+
+def getSupportedThermostatModes() {
+	modes()
+}
+
+def getMinSetpointIndex() {
+	0
+}
+def getMaxSetpointIndex() {
+	1
+}
+def getThermostatSetpointRange() {
+	(getTemperatureScale() == "C") ? [5, 30] : [41, 86]
+}
+def getHeatingSetpointRange() {
+	thermostatSetpointRange
+}
+
+def getSetpointStep() {
+	(getTemperatureScale() == "C") ? 0.5 : 1.0
+}
+
+def setupHealthCheck() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def configureSupportedRanges() {
+	sendEvent(name: "supportedThermostatModes", value: supportedThermostatModes, displayed: false)
+	sendEvent(name: "thermostatSetpointRange", value: thermostatSetpointRange, displayed: false)
+	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
+}
+
+def installed() {
+	setupHealthCheck()
+
+	configureSupportedRanges()
+}
+
+def updated() {
+	setupHealthCheck()
+
+	configureSupportedRanges()
+
+	unschedule(scheduledUpdateWeather)
+	if (settings.zipcode) {
+		runEvery1Hour(scheduledUpdateWeather)
+		scheduledUpdateWeather()
+	}
+}
+
+def parse(String description) {
+	if (description == "updated") {
+		return null
+	}
+
+	// Class, version
+	def map = createEvent(zwaveEvent(zwave.parse(description, [0x40:2, 0x43:2, 0x31:3, 0x42:1, 0x20:1, 0x85: 2])))
+	if (!map) {
+		return null
+	}
+
+	def result = [map]
+	if (map.isStateChange && map.name in ["heatingSetpoint","thermostatMode"]) {
+		def map2 = [
+			name: "thermostatSetpoint",
+			unit: getTemperatureScale(),
+			data: [thermostatSetpointRange: thermostatSetpointRange]
+		]
+		if (map.name == "thermostatMode") {
+			state.lastTriedMode = map.value
+			map2.value = device.latestValue("heatingSetpoint")
+		}
+		else {
+			def mode = device.latestValue("thermostatMode")
+			log.debug "THERMOSTAT, latest mode = ${mode}"
+			if (map.name == "heatingSetpoint") {
+				map2.value = map.value
+				map2.unit = map.unit
+			}
```

Comment:
I think I removed a lot of the stuff that made it less "useless". But anyway, since this code is mostly to appease the "consolidated deprecated Thermostat capability" gods, I think I have a way to better clean it up.